src/Sylius/Bundle/ApiBundle/Validator/Constraints/EligibleProductVariantToCartValidator.php

```diff
@@ -14,59 +14,74 @@
 namespace Sylius\Bundle\ApiBundle\Validator\Constraints;
 
 use Sylius\Bundle\ApiBundle\Command\Cart\AddItemToCart;
+use Sylius\Bundle\ApiBundle\Command\Cart\ChangeItemQuantityInCart;
 use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
 use Sylius\Component\Core\Model\ProductInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
 use Sylius\Component\Core\Repository\OrderRepositoryInterface;
 use Sylius\Component\Inventory\Checker\AvailabilityCheckerInterface;
 use Sylius\Component\Product\Repository\ProductVariantRepositoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
 use Symfony\Component\Validator\Constraint;
 use Symfony\Component\Validator\ConstraintValidator;
 use Webmozart\Assert\Assert;
 
 /** @experimental */
-final class AddingEligibleProductVariantToCartValidator extends ConstraintValidator
+final class EligibleProductVariantToCartValidator extends ConstraintValidator
 {
     /** @var ProductVariantRepositoryInterface */
     private $productVariantRepository;
 
+    /** @var RepositoryInterface */
+    private $orderItemRepository;
+
     /** @var OrderRepositoryInterface */
     private $orderRepository;
 
     /** @var AvailabilityCheckerInterface  */
     private $availabilityChecker;
 
+    /** @var ProductVariantInterface|null */
+    private $productVariant;
+
+    /** @var string */
+    private $productVariantCode = '';
```

Comment:
We should try to avoid storing a state in class fields and defining not meaningful default values. I don't see any valid reason to extract this field. If you need this `assignProductVariantAndProductVariantCode` function, it would be better to return an array with two fields and assign them instantly to two variables. A better solution would be to even split this method by half and execute similar logic twice. Another step forward would be to add VO with both values. But the best and simplest solution would be to unpack this function into the main function flow and just forget about this new function